src/coreclr/src/tools/aot/ILCompiler.ReadyToRun/Compiler/DependencyAnalysis/ReadyToRun/MethodFixupSignature.cs

```diff
@@ -59,7 +59,7 @@ public override ObjectData GetData(NodeFactory factory, bool relocsOnly = false)
             if (!_method.Unboxing && !_isInstantiatingStub && _method.ConstrainedType == null &&
                 fixupKind == ReadyToRunFixupKind.MethodEntry)
             {
-                if (!_method.Method.OwningType.HasInstantiation && !_method.Method.OwningType.IsArray)
+                if (!_method.Method.HasInstantiation && !_method.Method.OwningType.HasInstantiation && !_method.Method.OwningType.IsArray)
                 {
                     if (_method.Token.TokenType == CorTokenType.mdtMethodDef)
                     {
```

Comment:
How did we ended with `mdtMethodDef` or `mdtMemberRef` token here for method entrypoint with instantiation?

I would expect entrypoints like that to always to `methodSpec` token.

Reply:
The problem is that the method is part of a generated PInvoke i.e. it's synthetic IL. It doesn't come from the original SPC MSIL and so we don't have any token to encode it; that's why the synthetic token logic comes up with the original generic method definition methodDef token as it's "the best chance" to encode the method.

Reply:
Sounds like a fallout of what was discussed here: https://github.com/dotnet/runtime/pull/1762#discussion_r366942589

I don't have an opinion on this - I guess we still have option to just inline `GetArrayDataReference` into the callsite.